### PR TITLE
Dark Mode: update colors for login with updated `WordPressAuthenticator`

### PR DESCRIPTION
**Pull request by jaclync**
Login colors for #1555 

- [x] ⚠️ update PR base to `release/3.2` after https://github.com/woocommerce/woocommerce-ios/pull/1564 is merged ⚠️ 

`WordPressAuthenticator` PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/164

## Changes
- Updated `WordPressAuthenticator` pod to the beta version - I will point it to the official version after the `WordPressAuthenticator` PR is approved and released
- Updated colors we pass to `AuthenticationManager`
- Used primary button style for the Log In button
- Updated colors in `StorePickerViewController`, and always present it fullscreen

## Testing

- Launch the app
- Log out
- Log in --> the colors should look as expected

## Example screenshots

Log in | Log in - highlighted | Log in with email | Log in with email - ready | Request Link | 2FA | store picker
-- | -- | -- | -- | -- | -- | --
Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 27 44 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 27 45 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 30 03 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 30 11 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 39 03 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 44 22 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 42 15
Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 27 33 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 27 36 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 30 25 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 30 30 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 39 09 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 44 33 | Simulator Screen Shot - iPhone 8 - 2019-12-06 at 10 42 01

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by pmusolino — CHANGES_REQUESTED** [review]
Code LGTM ✅ 

## UI Testing

Seems that some colors don't match and they are identical in dark/light mode. 
See screenshots below:

| Log in - highlighted     |  Log in with email |
:-------------------------:|:-------------------------:
highlighted light | email light  
highlighted dark |  email dark OK

**Review by kyleaparker — CHANGES_REQUESTED** [review]
- The button on the first sign in screen on dark mode should be white
- There is an additional background colour behind the log in button on dark mode that needs to be removed. the only colour should be from the diagonal background

**Comment by jaclync** [comment]
@kyleaparker @pmusolino I updated the screenshots in the PR description, and the code to fix iOS 12 colors and some colors/assets in the Log in screens for another look!

**Review by mindgraffiti — APPROVED** [review]
My tests showed consistent colors like the screenshots shared in the summary. Nice work!

✅ code
✅ UI tested

**Review by kyleaparker — COMMENTED** [review]
Looks good, the only problem I see is on the 2FA light mode, the status bar should use white text/icons

File: Podfile.lock [hunk 1]
```diff
@@ -1,7 +1,7 @@
 PODS:
   - 1PasswordExtension (1.8.5)
   - Alamofire (4.7.3)
-  - Automattic-Tracks-iOS (0.4.1):
+  - Automattic-Tracks-iOS (0.4.2):
```
Comment (astralbodies) [inline Podfile.lock]: It looks like this PR updated more than just the Authenticator pod based on the changes of the lockfile. With just the two pod changes you can issue:

`bundle exec pod update WordPressAuthenticator WordPressUI`

That way we're not introducing a bunch of other unexpected dependency changes. 😄

File: WooCommerce/Classes/Extensions/NSAttributedString+Helpers.swift [hunk 2]
```diff
@@ -10,32 +10,4 @@ extension NSAttributedString {
     var foundationString: NSString {
         return string as NSString
     }
-
-    /// Returns a copy of the receiver, with the Leading + Trailing newlines trimmed.
-    ///
-    func trimNewlines() -> NSAttributedString {
```
Comment (astralbodies) [inline WooCommerce/Classes/Extensions/NSAttributedString+Helpers.swift]: Was this function no longer used?

**Review by astralbodies — CHANGES_REQUESTED** [review]
Re-do the CocoaPods update and good to go from me!

File: Podfile.lock (hunk 1 above)
Comment (jaclync) [reply Podfile.lock]: it's interesting that `pod update WordPressAuthenticator WordPressUI` still updated the `Automattic-Tracks-iOS` pod. Eventually, I had to first set `Automattic-Tracks-iOS` to a fixed version (`pod 'Automattic-Tracks-iOS', '0.4.1'`), run `pod install` and then change it back to `pod 'Automattic-Tracks-iOS', '~> 0.4.0'` to revert the pod version. I was wondering if it's safer to set the pods to a fixed version for the ones we're not actively working on?

File: WooCommerce/Classes/Extensions/NSAttributedString+Helpers.swift (hunk 2 above)
Comment (jaclync) [reply WooCommerce/Classes/Extensions/NSAttributedString+Helpers.swift]: this function used to an internal function in `WordPressKit-iOS`, and it was recently made public for usage in the host app: https://github.com/wordpress-mobile/WordPressKit-iOS/commit/5505e41b5133f463e0156423d4c870321be7a068#diff-e07293832952daad30d7515bb8fe5823 and we got this update from the `WordPressAuthenticator` pod update (https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator.podspec#L42)  - the implementation we had looks like a cloned version of the one in `WordPressKit-iOS` 🙂

**Comment by jaclync** [comment]
> Looks good, the only problem I see is on the 2FA light mode, the status bar should use white text/icons

@kyleaparker we set the preferred status bar style to light content (Light content, for use on dark backgrounds) everywhere in the app (like in Login Navigation https://github.com/woocommerce/woocommerce-ios/blob/develop/WooCommerce/Classes/Extensions/LoginNavigationController+Woo.swift#L10). I'm guessing iOS might have their own logic to determine the status bar text color. I couldn't reproduce this issue and this could be an issue on the simulator when I'm switching Light/Dark mode on the same screen, I'll look out for it!

File: Podfile.lock (hunk 1 above)
Comment (jaclync) [reply Podfile.lock]: reverted in: https://github.com/woocommerce/woocommerce-ios/pull/1565/commits/8ffa7ea5204449b18847cc399fe34ff115c29268

**Comment by jaclync** [comment]
merging now after updating the PR and responding to the CR comments!

**Comment by kyleaparker** [comment]
>I'm guessing iOS might have their own logic to determine the status bar text color. I couldn't reproduce this issue and this could be an issue on the simulator when I'm switching Light/Dark mode on the same screen, I'll look out for it!

I was referencing the screenshot you posted for 2FA light mode, all the other light mode screens use white text in the status bar (when it's purple) except for that one

**Pull request merged**